.github/workflows/ci.yml

```diff
@@ -0,0 +1,29 @@
+name: ci
+on: push
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version: 'stable'
+      - uses: golangci/golangci-lint-action@v6
+        with:
+          version: latest
+      - name: mod tidy check
+        run: .github/scripts/go-mod-tidy-check.sh
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macOS-latest, windows-latest]
```

Comment:
in-depth!!!!!11

I guess we could save some bucks here? unless there are differences expected across runtimes of course

Reply:
was going with what I had put in https://github.com/graxinc/kmutex, which always seems nice to ensure there is nothing platform specific. Too expensive? :D

Reply:
ahh nice!

yeeea, thinking now I'm sure it's peanuts hehe carry on!